Throw error if files are missing.

MancunianSam opened the pull request:
What was happening if you ran this with a non existent consignment was it would try to get the files from the api and the api would return nothing.
This empty list was passed to the s3 downloader which would download nothing.
The program would continue until we had a bagit package with an empty data directory.
Arguably, the API should be rejecting non-existent consignments but for now, this will throw an error if no files are returned by the api.

suzannehamilton commented on exporter/src/main/scala/uk/gov/nationalarchives/consignmentexport/Main.scala:
```diff
@@ -27,6 +27,7 @@ object Main extends CommandIOApp("tdr-consignment-export", "Exports tdr files in
         s3Files = S3Files(S3Utils(s3Async))
 
         data <- graphQlApi.getFiles(config, consignmentId)
+        _ <- IO.fromOption(data.headOption)(new Exception(s"No files found for consignment $consignmentId"))
```
Really minor thing, but could this say something like "Consignment API returned no files for consignment $consignmentId"? If we ever hit this error, it will be helpful to know to check the DB/API first rather than S3 or EFS.

suzannehamilton reviewed (approved):
Cheers for this. I think you're right that export should fail with an error if the data looks like it's in a bad state.

suzannehamilton reviewed (changes requested):
Sorry, hit Approve when I meant to hit Comment. It would be good to make the error message more specific.

MancunianSam commented:
The error you've suggested makes sense, I've changed it. I also realised that there are no tests for that error so I've added in a test to MainSpec to check for it.
